Orange/widgets/visualize/owscatterplot.py

```diff
@@ -368,7 +375,27 @@ def update_attr(self, attributes=None):
         self.cb_class_density.setEnabled(self.graph.can_draw_density())
         self.send_features()
 
+    def update_shapes(self):
+        if (self.cb_attr_shape.count() > 1 and
+            self.combo_value(self.cb_attr_shape) == None):
+                self.warning("Please select a shape attribute")
+        else:
+            self.warning()
+        self.graph.update_shapes()
+
+    def update_sizes(self):
+        if (self.cb_attr_size.count() > 1 and
+            self.combo_value(self.cb_attr_size) == None):
+                self.warning("Please select a size attribute")
```

Comment:
This segment looks awfully duplicated (three times). Could we put this check in a single place?